### ISSUE TITLE
Happy to have found this project, but questions

### ISSUE THREAD

**Issue by ShamanTcler** [mwganson/joint#2]
When I try a tenon I get something that has rounded ends, what if I wanted square ends? Something like a craftman style tenon?

Also I don't see the option for a "reveal". Think of it this way the rule of thirds ... so if you have a 5 unit (in or mm) face with a 1 unit reveal the thickness is 5- 2*1 or 3 ... does this make sense?

I have done a mortise and tenon joint using sketches and spreadsheets, but that was before finding your work.

Regards
Carl

**Comment by mwganson** [comment]
I don't think it would be difficult to add a Square option.  I'm thinking a boolean property named Square.  If True, you get the square ends for the mortise and tenon joint types.

I don't understand the reveal concept.  How would it work in terms of existing property names?  Make a default tenon on the front face of a default Part::Cube primitive.
![Snip macro screenshot-e8c536](https://user-images.githubusercontent.com/39143564/157294589-21b5b2d9-84b1-4e98-a4f4-b343967d7ff1.png)

I envision a floating point property named Reveal in the Dimensions group.  It would default to 0.0.  Reveal = 1.0 changes the other properties how?  Reveal = 2.0?  Would Reveal = -1 make sense?  I presume if Reveal is not equal zero, then some of the other properties become calculated based on one of the others?  Then those would need be made readonly while Reveal is non-zero?

**Comment by ShamanTcler** [comment]
How would you feel about me making a branch?

I am a retired Mechanical Engineer who wrote CAD/CAE for 40 years in C++ and last couple Python.  I just want to make sure it was Ok.

Sending you a private message.

**Comment by ShamanTcler** [comment]
From: https://www.woodworkersjournal.com/mortise-tenon-know-parts/  for a bit of consistent naming.

And https://www.woodmagazine.com/woodworking-how-to/joinery/mortise-tenon-joinery/rule-of-thumb-for-mortise-and-tenon-size  for more guidance

What I was calling reveal they call "Cosmetic Shoulder"

From looking at the code, I don't see a way to mate a mortise to a tenon. ( as in a name linkage ) So that if the mortise dimension C is 0.003" (0.07mm) larger than C on the tenon .... yes I am looking at using the Path workbench to cut these on my CNC

**Comment by mwganson** [comment]
They are created as separated objects.  Properties may be linked using FreeCAD's Expression engine.  For example, to link the Width property of Joint001 to the Width property of Joint, select the Width property of Joint001, press the = key (or click the f(x) icon) and enter Joint.Width+.07.  Then any time Joint.Width is changed Joint001.Width will change with it.

**Comment by mwganson** [comment]
I added a Rounded property for mortise/tenon joint types.

**Comment by ElijahLynn** [comment]
> How would you feel about me making a branch?
> 
> I am a retired Mechanical Engineer who wrote CAD/CAE for 40 years in C++ and last couple Python. I just want to make sure it was Ok.
> 
> Was looking for a way to send a PM to take the conversation out of the public

public discussions benefit everyone, so I encourage you to do it the "opensource way" and keep the discussions open too! Welcome to opensource!

**Comment by mwganson** [comment]
Go ahead and make a branch if you like.
